src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -23,11 +23,48 @@
 #include <app/util/af-event.h>
 #include <app/util/af.h>
 #include <app/util/attribute-storage.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/PlatformManager.h>
 
 using namespace chip;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::ColorControl;
 
+/**********************************************************
+ * Matter timer scheduling glue logic
+ *********************************************************/
+
+void ColorControlServer::timerCallback(System::Layer *, void * callbackContext)
+{
+    auto control = static_cast<EmberEventControl *>(callbackContext);
+    (control->callback)(control->endpoint);
+}
+
+void ColorControlServer::scheduleTimerCallbackMs(EmberEventControl * control, uint32_t delayMs)
+{
+    CHIP_ERROR err = DeviceLayer::SystemLayer().StartTimer(chip::System::Clock::Milliseconds32(delayMs), timerCallback, control);
+
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "Color Control Server failed to schedule event: %" CHIP_ERROR_FORMAT, err.Format());
+    }
```

Comment:
On success, doesn't this function need to set `control->status = EMBER_EVENT_MS_TIME` to be compatible with what the old code did?

As a followup (separate PR) we can look into getting rid of the `status` member, I think.....